Website/admin/Skins/Login.ascx.cs

```diff
@@ -145,11 +145,20 @@ protected override void OnLoad(EventArgs e)
 				        }
 
 				        string returnUrl = HttpContext.Current.Request.RawUrl;
-				        if (returnUrl.IndexOf("?returnurl=") != -1)
-				        {
-					        returnUrl = returnUrl.Substring(0, returnUrl.IndexOf("?returnurl="));
-				        }
-				        returnUrl = HttpUtility.UrlEncode(returnUrl);
+                        const string returnUrlToken = "?returnurl=";
+
```

Comment:
I would like to see the string manipulation on this control updated to use the overload that doesn't require a ToLowerInvariant call.  The use of ToLower() etc has some memory implications and additionally the lack of using a conversion or case insensitive comparison could impact URL's for the initial match.